core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java

```diff
@@ -978,6 +978,10 @@ public void addTaskToQueue(Task task) {
     public void executeSystemTask(WorkflowSystemTask systemTask, String taskId, int unackTimeout) {
         try {
             Task task = executionDAOFacade.getTaskById(taskId);
+            if (task == null){
```

Comment:
Wondering why executionDAO would return null instead of throwing error in the first place.